guides/micronaut-mongodb-asynchronous/java/src/main/java/example/micronaut/MongoDbFruitRepository.java

```diff
@@ -0,0 +1,44 @@
+package example.micronaut;
+
+import com.mongodb.client.result.InsertOneResult;
+import com.mongodb.reactivestreams.client.MongoClient;
+import com.mongodb.reactivestreams.client.MongoCollection;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+@Singleton // <1>
+public class MongoDbFruitRepository implements FruitRepository {
+
+    private final MongoDbConfiguration mongoConf;
+    private final MongoClient mongoClient;
+
+    public MongoDbFruitRepository(MongoDbConfiguration mongoConf,  // <2>
+                                  MongoClient mongoClient) {  // <3>
+        this.mongoConf = mongoConf;
+        this.mongoClient = mongoClient;
+    }
+
+    @Override
+    public Mono<Boolean> save(@NonNull @NotNull @Valid Fruit fruit){
+        return Mono.from(getCollection().insertOne(fruit)) // <4>
+                .map(insertOneResult -> Boolean.TRUE)
+                .onErrorReturn(Boolean.FALSE);
+    }
```

Comment:
Same comment as in the Mongo Sync guide. I think it is a common use case to have an `id` field assigned automatically Mongo and then return the `Fruit` object with that `id` field updated.